### PR TITLE
9956: do not test `contains` result against S.true 

### PR DESCRIPTION
**Pull request by smichr**
The test added identifies and issue with testing with "is true" instead of "== True" in the sets module. 
In master, the Intersection gives EmptySet because geometry returns True rather than true.

I have tried to change all such cases in sets to use True/False instead of assume that the `true` value will be returned by any comparison.

### CODE REVIEW

**Comment by smichr** [comment]
ping @gxyd

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ def _eval_imageset(self, f):
             result = S.EmptySet
             domain_set = self
             for (p_expr, p_cond) in expr.args:
-                if p_cond is S.true:
+                if p_cond is true:
```
Comment (gxyd) [inline sympy/sets/sets.py]: Any reason for not using `==` here?

File: sympy/sets/tests/test_sets.py [hunk 2]
```diff
@@ -453,6 +453,12 @@ def test_contains():
     assert RootOf(x**5 + x**3 + 1, 0) in S.Reals
     assert not RootOf(x**5 + x**3 + 1, 1) in S.Reals
 
+    # non-bool results
+    assert Union(Interval(1, 2), Interval(3, 4)).contains(x) == \
```
Comment (gxyd) [inline sympy/sets/tests/test_sets.py]: These tests work the same in master. Do these test something different?

File: sympy/sets/tests/test_sets.py [hunk 3]
```diff
@@ -453,6 +453,12 @@ def test_contains():
...
+        Or(And(x <= 2, x >= 1), And(x <= 4, x >= 3))
+    assert Intersection(Interval(1,x),Interval(2,3)).contains(y) == \
```
Comment (gxyd) [inline sympy/sets/tests/test_sets.py]: Perhaps space `1,x` -> `1, x` and `2,3` -> `2, 3` could be added

File: sympy/sets/sets.py (hunk 1 above)
Comment (smichr) [inline sympy/sets/sets.py]: Piecewise sympifies its args so any True is converted to S.true:

```
>>> Piecewise((x,x>0),(1,True))
Piecewise((x, x > 0), (1, True))
>>> _.args
((x, x > 0), (1, True))
>>> type(_[1][1])
<class 'sympy.logic.boolalg.BooleanTrue'>
```

File: sympy/sets/tests/test_sets.py (hunk 2 above)
Comment (smichr) [inline sympy/sets/tests/test_sets.py]: No, they just establish that no change should lose these elements by, perhaps, just testing for true and false values: one must account for the Basic case.

File: sympy/sets/tests/test_sets.py (hunk 3 above)
Comment (smichr) [inline sympy/sets/tests/test_sets.py]: done -- sorry I missed that.

**Comment by gxyd** [comment]
You have merges conflicts here. The PR looks good to me.

**Pull request merged**